main.go

```diff
@@ -34,30 +50,63 @@ func main() {
 	gitlabToken := getEnvOrError(GitlabToken)
 	gitEmail := getEnvOrError(GitEmail)
 	tfParallelism := getEnvOrDefault(TfParallelism, "10")
+	logFlushDelay := getEnvOrDefault(LogFlushDelay, "3")
 
 	tfParallelismInt, err := strconv.Atoi(tfParallelism)
-
 	if err != nil {
 		log.Fatal("Integer value required for `TF_PARALLELISM` environment variable")
 	}
 
-	err = pkg.Run(cfgPath,
-		workdir,
-		vaultAddr,
-		roleID,
-		secretID,
-		gitlabLogRepo,
-		gitlabUsername,
-		gitlabToken,
-		gitEmail,
-		tfParallelismInt,
-	)
+	logFlushDelayInt, err := strconv.Atoi(logFlushDelay)
 	if err != nil {
-		log.Fatalln(err)
+		logFlushDelayInt = 3
+	}
+
+	// Run main logic in goroutine
+	go func() {
+		err := pkg.Run(cfgPath,
+			workdir,
+			vaultAddr,
+			roleID,
+			secretID,
+			gitlabLogRepo,
+			gitlabUsername,
+			gitlabToken,
+			gitEmail,
+			tfParallelismInt,
+		)
+		done <- err
+	}()
+
+	// Wait for completion or signal
+	select {
+	case err := <-done:
+		if err != nil {
+			log.Printf("Error: %v [%s]", err, sessionID)
+			performGracefulShutdown(sessionID, logFlushDelayInt)
+			os.Exit(1)
+		}
+		log.Printf("Completed successfully [%s]", sessionID)
+	case sig := <-sigChan:
+		log.Printf("Received signal %v, shutting down [%s]", sig, sessionID)
```

Comment:
when we receive SIGTERM, do we need to pass it down to `pkg.Run` to let it also graceful shutdown?

Reply:
I removed the graceful shutdown and just added a slight delay before exiting instead of worrying about passing signals to goroutines. I've also identified some memory issues on the cluster with the collector pods that I aim to fix up.